### DISCUSSION TITLE
Result interpretation in vignette6

### DISCUSSION THREAD

**Discussion by hilmanamin** [JenniNiku/gllvm#118] [Q&A] [answered]
Hi.. vignettte6 gives an interesting study about predictor variables to multivariate community data. how we can interpret the result in `summary(CGLLVM)`? some coefficient estimates give negative values and some are positive, for example, soil.dry(LV2)        1.5032 and fallen.leaves(LV1)  -0.6957. Is it can be concluded that most spider species prefer in more dry soil and they did not prefer in many fallen leaves habitats?

Thanks.

**Comment by BertvanderVeen** [comment, marked as answer]
Thanks for posting you question!

In general, I would be cautious with inferring on the effect of predictor slopes for specific latent variables. 

In the example, the slopes for soil dry mass are both positive. In this case, I might interpret that as indicator of a positive relation between spider abundance and soil dry mass, though I am not an expert on spiders nor on soil dry mass! Soil dry mass reflects the organic matter content of soils, and is not a reflection of moisture content. It might, however, be an indicator of how productive a soil is.

Similarly with fallen leaves, but now we have a predictor that might reflect a measure of cover for hunting spiders. The slope is negative on both latent variables, so that there the model seems to indicate that there is a negative relationship between hunting spider abundance and places with cover? That does seem rather unintuitive, but I'm not a spider expert!

If you are interested in species-specific effects, those can (for a linear response model as is the case here) be retrieved as `CGLLVM$params$LvXcoef%*%t(CGLLVM$params$theta)`, since these effects are likely to differ on a species-specific basis. Alternatively, you can have a look at the ordination plot to see how species relate to these effects `ordiplot(CGLLVM, biplot = TRUE)`.

I hope that answers your question.

**Comment by hilmanamin** [comment]
Thanks for your great explanations. I think the inference from ordiplot was more favorable. But, when I use my dataset, is it any explanation for why the result always changes if I rerun the analysis? is it because I set inappropriate `lv.formula` ? for example I just define `lv.formula = ~DO + salinity + pH + temperature`. and I observe that applying `family="negative.binomial"` produce a more inconsistent result than `family="poisson"`. Is it must use `family="poisson"` in here?. I have no issue with using NB in mvabund package.
Also, is it possible to put a legend of environmental predictor gradient in ordiplot (for example legend for pH in vignette2)?

**Comment by BertvanderVeen** [comment]
Unlike in classical constrained ordination methods, the algorithm used here is sensitive to the initial values. In case you know non-metric multidimensional scaling, think of it like that: you will have to re-run the analysis to find an optimal solution. 

Jenni has written a paper on this, but there are some instructions in the vignettes and help pages too. Once you have found an optimal solution (use e.g. the `n.init` argument), the seed is stored in `model$seed` so you can reproduce your model, without having to re-run it multiple times frequently.

You can add a legend to your ordination plot by using the `legend` function in for base R plotting.

**Comment by BertvanderVeen** [comment]
Details on the method are available in our preprint: https://t.co/XgW6qTR81w?amp=1.

**Comment by hilmanamin** [comment]
Great. thanks for sharing the preprint and explanations.
